js/hview.js

```diff
@@ -632,59 +670,73 @@ $(function () {
+               methods can't be prepared off the main thread. Also, aside from WebWorkers, Javascript in the
+               browser doesn't provide any alternative methods of multi-threaded programming. */
+            function processRootNodes(processedIndex /* Integer */, isNewRoundOfProcessing /* Boolean */) {
+                if (isNewRoundOfProcessing) {
+                    processingStartTime = Date.now()
+                }
 
-            for(let i = 0; i < rootNodes.length; i++) {
-                const tBox = $("<div>")
-                const tVList = $("<div>")
+                const tBox = divProtoType.cloneNode()
+                const tVList = divProtoType.cloneNode()
 
-                renderList(rootNodes[i], tVList, tBox, vListJQueries.length)
+                renderList(rootNodes[processedIndex], tVList, tBox)
                 vListJQueries.push(tVList)
                 boxJQueries.push(tBox)
+                addToNodeMap(rootNodes[processedIndex], processedIndex)
 
-                if ($("#vlist_content label").length == 0) {
-                    tVList.children().appendTo("#vlist_content")
-                    tBox.children().appendTo("#border-box")
-                    applyResizeData();
+                if (processedIndex == 0) {
+                    moveChildNodes(document.getElementById("vlist_content"), tVList)
+                    moveChildNodes(document.getElementById("border-box"), tBox)
+                    onFirstViewHierarchyRendered()
                 }
-            }
-            document.getElementById("tl-range").max = (rootNodes.length-1).toString()
-        })
 
-        let nodeMap /* Map<String, ViewNode[]> \ null */
-
-        function migrateSelectedState(index /* Integer */) {
-            if (nodeMap == null) {
-                nodeMap = new Map()
-                function constructNodeMap(nodes /* ViewNode[] */, rootNodeIndex /* Integer | null */ = null) {
-                    const noRootNodeIndexProvided = rootNodeIndex == null
-                    for (let i = 0; i < nodes.length; i++) {
-                        if (noRootNodeIndexProvided) {
-                            rootNodeIndex = i
-                        }
-                        let mapValue /* ViewNode[] | null */ = nodeMap.get(nodes[i].name)
-                        if (mapValue == null) {
-                            mapValue = Array(rootNodes.length)
-                            nodeMap.set(nodes[i].name, mapValue)
-                        }
-                        mapValue[rootNodeIndex] = nodes[i]
-                        constructNodeMap(nodes[i].children, rootNodeIndex)
+                if (processedIndex < rootNodes.length-1) {
+                    if ((Date.now() - processingStartTime) < ALLOWED_PROCESSING_TIME) {
+                        processRootNodes(processedIndex+1, false)
+                    } else {
+                        setTimeout(processRootNodes, 6, processedIndex+1, true);
```

Comment:
```
setTimeout(
    processRootNodes, 
    Math.max (16 + processingStartTime - Date.now(), 1),
    processedIndex+1,
    true)
```

This ensure that the 16ms fame time is constant instead of 6ms gap (incase the last frame took longer that 10ms)

Reply:
Good idea Sunny! Done.